reducer/src/main/java/com/graphicsfuzz/reducer/reductionopportunities/GlobalVariableDeclToExprReductionOpportunities.java

```diff
@@ -46,4 +83,41 @@ public GlobalVariableDeclToExprReductionOpportunities(TranslationUnit tu,
     finder.visit(tu);
     return finder.getOpportunities();
   }
+
+
+  @Override
+  public void visitVariablesDeclaration(VariablesDeclaration variablesDeclaration) {
+    super.visitVariablesDeclaration(variablesDeclaration);
+    if (!context.reduceEverywhere()) {
+      // Replacing initializers with a new assignment statement might change semantics,
+      // do not consider these reduction opportunities if we are not reducing everywhere.
+      return;
+    }
+
+    // As constant must always be initialized, we will not consider global variable declarations
+    // that have const qualifier (i.e., const int a = 1).
+    if (atGlobalScope() && !variablesDeclaration.getBaseType().hasQualifier(TypeQualifier.CONST)) {
+      globalVariableDecl.add(variablesDeclaration);
+    }
+  }
+
+  @Override
+  public void visitFunctionDefinition(FunctionDefinition functionDefinition) {
+    super.visitFunctionDefinition(functionDefinition);
+    // We consider only the global variable declarations with the initializer that have been
+    // found before main function.
+    if (functionDefinition.getPrototype().getName().equals("main")) {
```

Comment:
There's no need to have the explicit indices - as discussed in the comments earlier, there's no need to - and indeed no point - enforcing an ordering in which the opportunities are added.  So you should change this to:

for (VariablesDeclaration variablesDeclaration : globalVariableDecl) {
   for (VariableDeclInfo variableDeclInfo : variablesDeclaration.getDeclInfos()) {
      if (...) {
        ...
      }
   }
}

It's best to avoid stuff with explicit indices where possible.

Reply:
Since in the [previous PR](https://github.com/google/graphicsfuzz/pull/561/files#diff-901b1045d425182b8e372343e60e8227R179) there is a test checking the order of the variable assignment, if we do not enforce the order of the assignment here test will fail. So should I remove such test in this PR?

Reply:
Ah, I suggest you change that test so that it works for the ordering that the above reformulation will yield.